reccmp/isledecomp/compare/db.py

```diff
@@ -377,26 +397,28 @@ def _find_potential_match(
         # But this index will not help if we are checking for NULL, so we exclude it
         # by adding the plus sign (Reference: https://www.sqlite.org/optoverview.html#uplus)
         if match_decorate:
-            sql = """
-            SELECT recomp_addr
-            FROM `symbols`
-            WHERE +orig_addr IS NULL
-            AND decorated_name = ?
-            AND (compare_type IS NULL OR compare_type = ?)
-            LIMIT 1
+            # TODO: Change when/if decorated becomes a unique column
```

Comment:
Is this something you keep track of? I wouldn't know what to make of it when I read this line.

Reply:
That note is a leftover from earlier designs. We could use the symbol (i.e. linker name, or "decorated name" in MSVC land) as another unique ID alongside the two addresses. The idea in this function is that a potential match using a symbol name should skip all these checks on non-unique fields. (The DB should optimize the query plan to do that anyway.)

I took this out because multiple symbols can point to the same address. Right now we deal with that by throwing away the other symbol(s). This really only comes up in the C runtime library with string functions, but it's something to revisit for programs with handwritten assembly.